tests/ssl-opt.sh

```diff
@@ -2201,7 +2201,7 @@ run_test    "Opaque key for server authentication: ECDHE-RSA" \
 requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_2
 requires_config_enabled MBEDTLS_USE_PSA_CRYPTO
 requires_config_enabled MBEDTLS_X509_CRT_PARSE_C
-requires_config_enabled MBEDTLS_ECDSA_C
+requires_config_enabled MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED
```

Comment:
Note: ideally, this one could be auto-detected based on `force_ciphersuite=TLS-DHE-RSA-WITH-AES-128-CBC-SHA`. I'm OK with having it as a manual dependency for now, but at some point in the future it should be auto-detected.

(Arguably we could also auto-detect thinks like `"Ciphersuite is TLS-DHE-RSA"` but that would require looking beyond the server and client command lines, so that sounds like a deeper change.)

(Not asking for any change here, unless you feel like adding auto-detection of key exchange dependencies based on `force_ciphersuite` because it might help when you work on ECDH once you're done with ECDSA.)

Reply:
> unless you feel like adding auto-detection of key exchange dependencies based on `force_ciphersuite` because it might help when you work on ECDH once you're done with ECDSA

I tried to address this suggestion. Please check [this](https://github.com/Mbed-TLS/mbedtls/pull/7082/commits/133f9b056d0397aec3ee399b7c0dc792deb52db2) commit